### PR TITLE
Added quest details to pending quest display, tabs to pending/active quest display, and links to user profiles from quest lists

### PR DESCRIPTION
**Pull request by hairlessbear**
Before:
selection_008

After:
selection_009
selection_010

### CODE REVIEW

**Comment by lemoness** [comment]
I'm not sure about this.... I like the fact that the picture only appears
when we've started the quest, as it's a good visual indicator.

Thoughts from others?

**Comment by crookedneighbor** [comment]
I agree. It also pushes down the list of users with their invitation status as well as the accept and decline buttons.

**Comment by AnitaYCheng** [comment]
I would like a link to the quest pic/description in the title area, so it's there if people want to see that. But I agree, in pre-quest time, it's more useful to see the party member's invitation status.

**Comment by hairlessbear** [comment]
Hmm...I really want some way for the users to be able to check out the quest info without having to go back to the inventory page or look it up on the wiki. I like @AnitaYCheng's suggestion. Could we add a link that shows a modified version of the buy-a-quest pop-up (identical but without the footer)?

**Comment by Alys** [comment]
I like this change. I've often thought that the quest info should be in the pending box. We could put the info at the bottom of the box if we didn't want to push down the user list and buttons. Also maybe the image could be removed if we wanted the box to take up less space.

**Comment by crookedneighbor** [comment]
What about this:
screen shot 2015-05-06 at 9 06 14 pm

screen shot 2015-05-06 at 9 06 18 pm

**Comment by hairlessbear** [comment]
@crookedneighbor The problem I have with that solution is it removes all of the best parts of the quest (namely, the picture and flavor text). I could be alone on this, but for me, including the picture and flavor text is a high priority because they're what make each quest unique. Without them, the quests are reduced to a generic "Collect X/Do X damage until you win", and it's hard to look forward to each individual quest when they all look the same.

**Comment by AnitaYCheng** [comment]
I have a question: is the invitation status list important to anyone else in the party, aside from the quest leader? Presumably they need to see that, so they know when it's an appropriate time to start the quest. But for anyone else, I imagine they would be more interested in seeing what the quest entails, than seeing who's signed up for it.

Going off of that, would it make sense to make the invitation status list a popover? Then the quest pic (with a low opacity if it hasn't started yet) could be visible by default, with the description.

**Comment by crookedneighbor** [comment]
@AnitaYCheng Personally, I think the invitation list is important to the whole party, because it allows the whole party to keep those who have not accepted yet accountable to accept. If it was in a popover, it wouldn't be as obvious. 

That said, I do like your idea to give the quest art a slightly transparent look.

**Comment by deilann** [comment]
Agreed with @crookedneighbor -- even when I'm not the quest leader, I'll
often go to check to see who has and hasn't accepted the quest yet. I'd be
rather annoyed if I couldn't.

**Comment by Alys** [comment]
I agree about the importance of the invitation list to all party members, and also with @hairlessbear's [opinion](https://github.com/HabitRPG/habitrpg/pull/5157#issuecomment-99940520) about the picture and flavour text making each quest unique.

What do you all think about the idea of including the image and text underneath the invitation list and buttons? I see that as the best of both worlds.

**Comment by crookedneighbor** [comment]
I think that would be a good compromise. There's a `.transparent` class that gives an opacity of .5 and I think that should be applied to the quest image before it is actually begun:

screen shot 2015-05-08 at 7 50 10 am

**Comment by hairlessbear** [comment]
@crookedneighbor If the quest image and text are underneath the "Begin Quest" text/buttons, is it still necessary to add opacity? The fact that you have to scroll past the "Begin Quest" text/buttons seems like enough of an indicator to me.

**Comment by AnitaYCheng** [comment]
What about a segmented tab, like what we use to filter tasks?

A super-rough idea I just whipped up in Photoshop:

questinfo

questinvites

**Comment by Alys** [comment]
But what is wrong with putting the quest details under the invite list and
begin/cancel buttons? It's simple, requires minimal changes, and doesn't
depend on the users noticing the buttons.

**Comment by AnitaYCheng** [comment]
The way it is now, you have to scroll all the way down to see the details. (We're in a large party - you have to scroll quite a bit!) For the longest time, I didn't even know there was anything under the long list. If it was under the list but _not_ in the scrolling window, that would work okay for me. Because right now, you do have to scroll all the way down past the list to see the description of how quests work, and putting the quest description there would not make a difference.

**Comment by hairlessbear** [comment]
I really like @AnitaYCheng's idea of using tabs (I'm also in a large party, and agree that putting the info under the invitation list is cumbersome). I implemented a quick version of it, thoughts? 

selection_011

selection_013
(the above picture is here to show that Ian shows up regardless of what tab you're viewing)

selection_012

**Comment by Alys** [comment]
@hairlessbear I like it! However I think the list of players should be the default tab, because the most important thing is getting everyone to accept or reject.

**Comment by AnitaYCheng** [comment]
Looks great @hairlessbear! I agree with @Alys on the order. The ordering of my mockups was backwards. I think the image of the boss is a good visual indicator that a quest is on, and very difficult to mistake with an invite list, but the info is there in a convenient spot.

**Comment by hairlessbear** [comment]
@Alys, I'll make that change right now! Do you want me to switch the order of the buttons to match @AnitaYCheng's mock-up, or is it ok as is?

Also, while working on this issue, I noticed that while the quest is in progress, the quest details and rewards are separated by the participants list, which doesn't really make sense to me. Would it be ok if I positioned the rewards directly under the quest details? That would make the resulting order (top down) quest picture, progress, text, rewards, participant list.

**Comment by AnitaYCheng** [comment]
@hairlessbear switching the order of the buttons so that the default button (invite list) is on the left would be good.

How about the same treatment when the quest is in progress: A tab that contains the participant list? Just switch the buttons around when a quest is active vs not active. (Is that too confusing?) As a healer, I do want to regularly check the health of the participants, since we do take more damage than usual when questing.

An how about a link to the user's profile from that list? :wink: (Pretty please with a :cherries: :cake: ?)

**Comment by hairlessbear** [comment]
@AnitaYCheng I'll add the button behavior right now and post some updated screenshots! 

If we add links to the individual profiles from the participant list, I think they should also be added to the invitation list. This could be particularly useful when a user hasn't accepted the quest yet and others want to send them a message asking why. Only issue is I have no idea how to implement that, but I'll figure it out!

**Comment by Alys** [comment]
@hairlessbear Have a look at how the party members list does links.

**Comment by hairlessbear** [comment]
Since there's now a lot of information that's displayed regardless of the quest status, there's a lot of code duplication. I'm trying to cut it down via multiple conditional statements, but I quickly ran into an issue that I couldn't solve, and I'd greatly appreciate some help.

I tried to add an `ng-if` to the pills to change their text based on whether the quest was active or not. 

``` jade
ul.nav.nav-pills
  li(ng-class="{active:selectedTab === 1}")
    a(ng-if="group.quest.active==false" ng-click="selectedTab=1")
      ="Invitation List"
    a(ng-if="group.quest.active==true" ng-click="selectedTab=1")
      ="Quest Details"
```

But although it changed the text as expected, it made the buttons non-operational, and I have no idea why. I've tried multiple variations (e.g., moving the `ng-if` up to the `li`), but the buttons would only work when I removed the `ng-if` entirely. Anyone know what's going on? Or a better way to do what I'm attempting?

**Comment by hairlessbear** [comment]
Alright, all behavior requested by @AnitaYCheng has now been added, but I wasn't sure what was meant by "switch the buttons around". I can't tell whether this means switch the default tab, or switch the button placement when a quest is in progress (i.e., quest details on left, participants on right). I opted for the second option, but I could change it easily if I guessed wrong :) 

There's still a lot of duplicate code that can be reduced, but that's a problem for tomorrow.

### Quest pending

selection_017
selection_018

### Quest in progress

selection_014
selection_015
selection_016

P.S. Thanks for the tip @Alys!

**Comment by hairlessbear** [comment]
After posting, I realized that moving the buttons around makes things unnecessarily confusing. The most recent commit modifies the behavior so only the default tab and button title change when the quest starts. The buttons stay in their original place.

Also, what do you guys think about the term "invites"? It doesn't feel quite right to me, but I can't think of a better word :confused:

**Comment by lemoness** [comment]
I think "Invitations" works well!

**Comment by hairlessbear** [comment]
@lemoness Done!

Regarding the code duplication I mentioned earlier, I think cutting it down like I was planning would actually make the code much less clear. It would greatly increase the number of `quest.active` checks and be far less readable. So I'm not going to try to cut down the code duplication, unless someone else requests it.

Here are the most recent screenshots:

## Quest pending

selection_021
selection_022

(Ian is under the boss text if you scroll down)

## Quest in progress

selection_027
selection_024
selection_028

Are there any other changes/features that are needed/requested?

**Comment by crookedneighbor** [comment]
Do these changes interfere with how a world boss looks?

**Comment by hairlessbear** [comment]
@crookedneighbor I did my best to make sure it didn't, but I just noticed a mistake. I'll fix it right now, then post a message explaining why the PR shouldn't interfere with the world boss behavior.

**Comment by hairlessbear** [comment]
@crookedneighbor If I understand correctly, the world boss logic will never run when `group.quest.active==false`, so there are no `tavern` checks in that block of code. Within the `group.quest.active==true` block, I maintained the `tavern` checks that were already there, and made my code fit into them. Specifically, when `tavern` is true:
1) [The pills and quest participants don't show up](https://github.com/hairlessbear/habitrpg/blob/pending_quest_display/website/views/options/social/boss.jade#L65).
2) The [world boss image](https://github.com/hairlessbear/habitrpg/blob/pending_quest_display/website/views/options/social/boss.jade#L94) shows up properly.
3) The [quest rewards](https://github.com/hairlessbear/habitrpg/blob/pending_quest_display/website/views/options/social/boss.jade#L129) don't display.
4) The [Ian text](https://github.com/hairlessbear/habitrpg/blob/pending_quest_display/website/views/options/social/boss.jade#L134) is changed to `tavernBossInfo`.
5) The [abort button isn't displayed](https://github.com/hairlessbear/habitrpg/blob/pending_quest_display/website/views/options/social/boss.jade#L147).

One more thing: when `tavern` and `group.quest.active` are true, `selectedTab` is still set and initialized to 2, but this doesn't change the world boss behavior. Without the pills, there is no method of changing `selectedTab`'s value, and the [relevant div](https://github.com/hairlessbear/habitrpg/blob/pending_quest_display/website/views/options/social/boss.jade#L91) contains `ng-show="selectedTab === 2"`, so it will still display properly.

**Comment by hairlessbear** [comment]
Quick explanation of the most recent commits: I added a `br` between the pills and the content, and the most recent commits remove this `br` when there is a world boss.

File: website/views/options/social/boss.jade [hunk 1]
```diff
@@ -5,23 +5,50 @@ mixin boss(tavern, mobile)
...
+            a(ng-click="selectedTab = 2")
+              =env.t('questDetails')
+        div(ng-show="selectedTab === 1")
+          br
+          table.table.table-striped(bindonce='group')
+            tr(ng-repeat='member in group.members track by member._id')
+              td.media
+                span(ng-if=':: group.quest.leader && group.quest.leader==member._id && isMemberOfGroup(group.quest.leader,group) && isMemberOfPendingQuest(group.quest.leader,group)') 
+                  |*&nbsp;
+                a.media-body
```
Comment (crookedneighbor) [inline website/views/options/social/boss.jade]: What are these media classes for?

File: website/views/options/social/boss.jade [hunk 2]
```diff
@@ -5,23 +5,50 @@ mixin boss(tavern, mobile)
...
     .panel-body.modal-fixed-height
-      div(ng-if='group.quest.active==false')
-        table.table.table-striped
-          tr(ng-repeat='member in group.members')
-            td
-              span(ng-if=':: group.quest.leader && group.quest.leader==member._id && isMemberOfGroup(group.quest.leader,group) && isMemberOfPendingQuest(group.quest.leader,group)')
-                |*&nbsp;
-              |{{::member.profile.name}}
-            td {{group.quest.members[member._id] === true ? env.t('accepted') : group.quest.members[member._id] === false ? env.t('rejected') : env.t('pending')}}
+      div(ng-if='group.quest.active==false' ng-init="selectedTab = 1")
```
Comment (crookedneighbor) [inline website/views/options/social/boss.jade]: I think selectedTab is too generic. It should be specific to the quest box so collisions don't happen with other things on scope with the same name. Also, why use numbers to designate what state is current? Wouldn't strings be more descriptive?

File: website/views/options/social/boss.jade (hunk 1 above)
Comment (hairlessbear) [inline website/views/options/social/boss.jade]: Nothing! I copied the code from [here](https://github.com/HabitRPG/habitrpg/blob/develop/website/views/options/social/group.jade#L80-87), and forgot to remove them (they're necessary in the original code, but not here). Good catch.

File: website/views/options/social/boss.jade (hunk 2 above)
Comment (hairlessbear) [inline website/views/options/social/boss.jade]: Numbers were the first thing I thought of, but you're right, strings would be better. I'll change that and commit.

File: website/views/options/social/boss.jade [hunk 3]
```diff
@@ -5,23 +5,50 @@ mixin boss(tavern, mobile)
...
-              |{{::member.profile.name}}
-            td {{group.quest.members[member._id] === true ? env.t('accepted') : group.quest.members[member._id] === false ? env.t('rejected') : env.t('pending')}}
+      div(ng-if='group.quest.active==false' ng-init="questBoxTab = 'invitesTab'")
+        ul.nav.nav-pills
+          li(ng-class="{active:questBoxTab === 'invitesTab'}")
+            a(ng-click="questBoxTab = 'invitesTab'")
+                =env.t('invitations')
+          li(ng-class="{active:questBoxTab === 'detailsTab'}")
+            a(ng-click="questBoxTab = 'detailsTab'")
+              =env.t('questDetails')
```
Comment (crookedneighbor) [inline website/views/options/social/boss.jade]: Let's put this all on one line:  `a(ng-click="questBoxTab = 'detailsTab'")=env.t('questDetails')`

I only like putting the `=env.t` bit on it's own line if there's complicated logic in the element.

Comment (hairlessbear) [inline website/views/options/social/boss.jade]: Done. I'll remember that for future reference.

**Comment by crookedneighbor** [comment]
@hairlessbear I've lost track of this. What's the status? Are we waiting on you or are you waiting on us?

**Comment by hairlessbear** [comment]
I'm waiting on you guys (and doing my best to not nag you :P)

**Comment by crookedneighbor** [comment]
I've brought this up with develop, made some changes (used angular-ui's tabsets instead of nav-pills), and did some major refactoring of the quest jade files. Because of the major refactor, I'm waiting on some other devs to look it over before merging it in. 

If you're interested, here are your changes adn mine combined: https://github.com/HabitRPG/habitrpg/compare/develop...crookedneighbor:hairlessbear-pending_quest_display

**Comment by hairlessbear** [comment]
Nice work! 

With this refactor, it would be much easier to add the quest accept/reject buttons requested in #5298 here instead of there. If they're added over there, it'll have to be immediately changed as soon as this is added. 

That being said, the accept/reject buttons will be using the `RSVPNeeded` field that's added in the other PR, so they'd have to be merged at the same time for proper functionality. Does that sound like a reasonable plan, or would you still prefer the buttons be added in the other PR?

**Comment by crookedneighbor** [comment]
Let's wait till this refactor is merged in before doing any more work on either of them.

**Comment by hairlessbear** [comment]
Roger that.

**Comment by crookedneighbor** [comment]
Notated toward your next tier.

**Pull request merged**